### PR TITLE
Fix #150 Prevents Maintainers checklist comment on non-fork branches

### LINKED ISSUE #150: Maintainers checklist is not relevant for non-fork branches
See https://github.com/oppia/oppia/pull/11084#issuecomment-719022838. While most contributors create PRs that use forks and need to have that checkbox ticked, it's implicit for all branches created directly on the repository. We shouldn't raise this error except for branches on forks.

### PR DESCRIPTION
**Pull request by gp201**
## Explanation

  Fixes #150: Compares the full name of the head (ex: testuser/Oppia) and base repo(ex: oppia/oppa)

## Checklist
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

**Comment by gp201** [comment]
@jameesjohn PTAL

File: lib/checkPullRequestTemplate.js [hunk 1]
```diff
@@ -26,6 +26,18 @@ const UNCHECKED_CHECKLIST = '- [ ]';
...
+/**
+ * This function checks if head and base of the PR are from the same
+ * repository.
+ *
+ */
+const compareHeadAndBase = (prHead, prBase) => {
+  var reRepoName = /^(.+?):/gm;
+  var prHeadRepo = reRepoName.exec(prHead);
+  reRepoName.lastIndex = 0;
+  var prBaseRepo = reRepoName.exec(prBase);
```
Comment (jameesjohn) [inline lib/checkPullRequestTemplate.js:38]: Can we test this without using regex?

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks, @gp201.
I have just one comment.

File: lib/checkPullRequestTemplate.js (hunk 1 above)
Comment (gp201) [reply lib/checkPullRequestTemplate.js]: DONE

**Comment by gp201** [comment]
@jameesjohn PTAL

File: lib/checkPullRequestTemplate.js [hunk 2]
```diff
@@ -162,7 +162,9 @@ const checkEssentialChecklist = (pullRequest) => {
         // Check if maintainer can actually modify the PR. This is because
         // it is possible for the maintainers to be able to modify the PR
         // but the PR author forgot to check it in the description.
-        if (!pullRequest.maintainer_can_modify) {
+        if (!pullRequest.maintainer_can_modify &&
+          !(pullRequest.head.repo.full_name ===
+            pullRequest.base.repo.full_name)){
```
Comment (jameesjohn) [inline lib/checkPullRequestTemplate.js:167]: Can you extract this into a variable like ` isPrFromFork = pullRequest.head... === pullRequest.base...`?
Then use that with the `maintainer_can_modify` for the comparison.

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks.
Some final comments.

File: lib/checkPullRequestTemplate.js (hunk 2 above)
Comment (gp201) [reply lib/checkPullRequestTemplate.js]: DONE

**Comment by gp201** [comment]
@jameesjohn PTAL

File: lib/checkPullRequestTemplate.js [hunk 3]
```diff
@@ -162,7 +162,10 @@ const checkEssentialChecklist = (pullRequest) => {
         // Check if maintainer can actually modify the PR. This is because
         // it is possible for the maintainers to be able to modify the PR
         // but the PR author forgot to check it in the description.
-        if (!pullRequest.maintainer_can_modify) {
+        var isPrFromFork = pullRequest.head.repo.full_name ===
+        pullRequest.base.repo.full_name;
```
Comment (jameesjohn) [inline lib/checkPullRequestTemplate.js:166]:
```suggestion
        const isPrFromFork =
          pullRequest.head.repo.full_name === pullRequest.base.repo.full_name;
```

**Review by jameesjohn — CHANGES_REQUESTED** [review]

File: lib/checkPullRequestTemplate.js (hunk 3 above)
Comment (jameesjohn) [reply lib/checkPullRequestTemplate.js:166]: Prefer const over `let` or `var` where possible.

**Comment by gp201** [comment]
@jameesjohn PTAL

File: lib/checkPullRequestTemplate.js (hunk 3 above)
Comment (gp201) [reply lib/checkPullRequestTemplate.js]: DONE

File: lib/checkPullRequestTemplate.js [hunk 4]
```diff
@@ -162,7 +162,9 @@ const checkEssentialChecklist = (pullRequest) => {
         // Check if maintainer can actually modify the PR. This is because
         // it is possible for the maintainers to be able to modify the PR
         // but the PR author forgot to check it in the description.
-        if (!pullRequest.maintainer_can_modify) {
+        const isPrFromFork =
+          pullRequest.head.repo.full_name === pullRequest.base.repo.full_name;
+        if (!pullRequest.maintainer_can_modify && !isPrFromFork){
```
Comment (jameesjohn) [inline lib/checkPullRequestTemplate.js:167]:
```suggestion
        if (!pullRequest.maintainer_can_modify && !isPrFromFork) {
```

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Just one nit.

File: lib/checkPullRequestTemplate.js (hunk 4 above)
Comment (gp201) [reply lib/checkPullRequestTemplate.js]: DONE

**Comment by gp201** [comment]
@jameesjohn PTAL

File: spec/checkPullRequestTemplateSpec.js [hunk 5]
```diff
@@ -473,4 +473,43 @@ describe('Pull Request Template', () => {
...
+    it('should assign PR author', () => {
+      expect(github.issues.addAssignees).toHaveBeenCalled();
+      expect(github.issues.addAssignees).toHaveBeenCalledWith({
+        issue_number: payloadData.payload.pull_request.number,
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        assignees: [payloadData.payload.pull_request.user.login],
+      });
+    });
+  });
```
Comment (jameesjohn) [inline spec/checkPullRequestTemplateSpec.js:514]: I don't think this block is required.

**Review by jameesjohn — CHANGES_REQUESTED** [review]

File: spec/checkPullRequestTemplateSpec.js (hunk 5 above)
Comment (gp201) [reply spec/checkPullRequestTemplateSpec.js]: DONE

**Comment by gp201** [comment]
@jameesjohn PTAL

**Review by jameesjohn — APPROVED** [review]
LGTM!
Thanks, @gp201!

**Pull request merged**